client/app/src/androidTest/java/com/example/criengine/AddABookTest.java

```diff
@@ -58,19 +60,61 @@ public void setUp() {
     }
 
     /**
-     * Test to see if a user can edit the fields of a new book before creating it.
+     * Test to see if the user is blocked from adding a book without entering in the fields.
+     */
+    @Test
+    public void saveWithNoInfoTest() {
+        solo.clickOnButton("Save and Add Photo");
+        assertTrue(solo.waitForText("Please Fill Out All Fields", 1, 1000));
+    }
+
+    /**
+     * Test to see if a user can edit the fields of a new book before creating it. No image attached.
+     */
+    @Test
+    public void addBookWithNoImageTest() {
+        solo.enterText((EditText) solo.getView(R.id.newBookTitle), "Mock Title");
+        solo.enterText((EditText) solo.getView(R.id.newBookDesc), "This is a new Description");
+        solo.enterText((EditText) solo.getView(R.id.newBookAuthor), "This is a new Author");
+        solo.enterText((EditText) solo.getView(R.id.newBookISBN), "This is a new ISBN");
+
+        solo.clickOnButton("Save and Add Photo");
+
+        assertTrue(solo.waitForText("Before you go...", 1, 1000));
+        solo.clickOnText("NOT NOW");
+
+        solo.assertCurrentActivity("Wrong Activity", RootActivity.class);
+
+        solo.clickInList(0);
+        assertTrue(solo.waitForText("Mock Title", 1, 1000));
+        solo.goBack();
+        cleanup();
```

Comment:
isn't this what @After is for?

Reply:
not every test has a book to be removed so it shouldn't go into the after